Lib/test/test_ensurepip.py

```diff
@@ -1,19 +1,64 @@
-import unittest
-import unittest.mock
-import test.support
+import contextlib
 import os
 import os.path
-import contextlib
 import sys
+import tempfile
+import test.support
+import unittest
+import unittest.mock
 
 import ensurepip
 import ensurepip._uninstall
 
 
-class TestEnsurePipVersion(unittest.TestCase):
+class TestPackages(unittest.TestCase):
+    def touch(self, directory, filename):
+        fullname = os.path.join(directory, filename)
+        open(fullname, "wb").close()
+
+    def test_version(self):
+        # Test version()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            self.touch(tmpdir, "pip-1.2.3b1-py2.py3-none-any.whl")
```

Comment:
Any reason for not using `pathlib` ?
```suggestion
            pathlib.Path(tmpdir, "pip-1.2.3b1-py2.py3-none-any.whl").touch()
```

Reply:
I prefer to avoid testing pathlib in test_ensurepip :-) I like low-level code :-D